### ISSUE TITLE
Time shift playback no longer updates the recording time 

### ISSUE THREAD

**Issue by AldaTheFirst** [xbmc/xbmc#19462]
### Describe the bug
Here is a clear and concise description of what the problem is:

During time shift playback, the recording time is no longer updated (worked in an older version). If the playback for example started after 10 minutes while the recording is still running and you then watch 20 minutes leads to the fact that the Screem shows 20:00 / 10:00 (I watched 20 minutes, but the total length is only 10 minutes) and the progress bar is at the end. If you continue to fast forward, playback will stop and when you a new start leads to start from the beginning. 

## Expected Behavior
Here is a clear and concise description of what was expected to happen:

Total length of recording should be actualized during time shift.

## Actual Behavior

The displayed total duration of the recording is always the same (time already recorded when starting playback, no update) 

## Possible Fix

This was working in a former release. I think there was accidentally removed a line?

### To Reproduce

1. Start a recording from television (in my case I use vdr as a backend)
2. Wait a few minutes
3. Start playback of the recording when the recording is still in progress 
4. Look at the progress bar and on the display of the recorded time 
5. Recorded time is always the same until you stop playback and start again.

### Debuglog

The debuglog can be found here:
There is no bebuglog

### Screenshots 
Here are some links or screenshots to help explain the problem:

## Additional context or screenshots (if appropriate)
Here is some additional context or explanation that might help:

### Your Environment
Used Operating system:

 - [ ] Android
 - [ ] iOS
 - [ ] tvOS
 - [ x] Linux
 - [ ] OSX
 - [ ] Windows
 - [ ] Windows UWP

 - Operating system version/name: osmc
 - Kodi version:

18.9 Media Center Kodi
Copyright (C) 2005-2020 Team Kodi - http://kodi.tv

*note: Once the issue is made we require you to update it with new information or Kodi versions should that be required.
Team Kodi will consider your problem report however, we will not make any promises the problem will be solved.*

**Comment by rainman74** [comment]
I can also confirm! However, the entire Kodi hangs when the bar has reached the end and you were previously in timeshift.

Kodi 19 Final, Android TV 9

**Comment by ksooo** [comment]
Could be related to the PVR addon you are using. No problems here using Tvheadend.

**Comment by ksooo** [comment]
> There is no bebuglog

 If you want meaningful assistance, please post a debug log. Without a log we can only guess what happens on your end.

**Comment by phunkyfish** [comment]
> I can also confirm! However, the entire Kodi hangs when the bar has reached the end and you were previously in timeshift.
> 
> Kodi 19 Final, Android TV 9
> 
> PVR IPTV Simple Client / current version

IPTV Simple does not support recordings. So I’m unsure how you could be having the same problem?

**Comment by phunkyfish** [comment]
@AldaTheFirst are you using Debian?

**Comment by phunkyfish** [comment]
Hold on, Kodi 18.9 is the previous release of Kodi. @AldaTheFirst please try version 19.1 of Kodi first and report back. If the issue persists please provide a full debug log and specify the exact linux platform and version you are using.

**Comment by AldaTheFirst** [comment]
Hi phunkyfish

I am using:

Linux vdr4 4.9.0-4-amd64 #1 SMP Debian 4.9.65-3 (2017-12-03) x86_64 GNU/Linux as PVR TV with

vdr (2.2.0/2.2.0) - The Video Disk Recorder
skinenigmang (0.1.2) - EnigmaNG skin
streamdev-server (0.6.1-git) - VDR Streaming Server
svdrposd (1.0.0) - Publish OSD menu via SVDRP
quickepgsearch (0.0.1) - Quick search for broadcasts
conflictcheckonly (0.0.1) - Direct access to epgsearch's conflict check menu
svdrpservice (1.0.0) - SVDRP client
xineliboutput (2.0.0-cvs) - X11/xine-lib output plugin
osdserver (0.1.3) - Server for remote OSD clients
epgsearch (1.0.1.beta5) - search the EPG for repeats and more
vnsiserver (1.5.1) - VDR-Network-Streaming-Interface (VNSI) Server
epgsearchonly (0.0.1) - Direct access to epgsearch's search menu

18.9 Media Center Kodi
Linux osmc 4.19.122-2-osmc #1 SMP PREEMPT Sat Jul 4 02:39:56 UTC 2020 armv7l GNU/Linux
Raspberry Pi 3 Model B Rev 1.2

Hope that will help.

Alda

**Comment by AldaTheFirst** [comment]
Also @phunkyfish :
I will try kodi 19.1, but need some time because the backup needs some time.

**Comment by phunkyfish** [comment]
> Also @phunkyfish :
> I will try kodi 19.1, but need some time because the backup needs some time.

Ok, you should wait for 19.1 on Debian if you also run it locally there. I think it is being released shortly @basilgello ?

**Comment by basilgello** [comment]
Yes, I am going to push everything tonight including unblock requests

**Comment by basilgello** [comment]
19.1 is on the way to Debian bullseye & sid.

**Comment by AldaTheFirst** [comment]
Hello,
Now I have uploaded the debug.log to https://paste.kodi.tv/oxahedumof
The behavior is a little different now:
For example:
The recording started at 2:00 p.m. and remains active all the time
Playback starts at 2:03 p.m.
The display shows 00:00 / 00:03
After 2 minutes the display shows 00:02 / 00:03, but should show 00:02 / 00:05
After another 2 minutes the display shows 00:04 / 00:03, means I saw more than the duration of the recording ? Could not be.
Pressing Jump Forward (ff) will reset the playback to 00:03 and start playing from that point
If you stop and restart playback, recording will start from the beginning of 00:00 

Hope, that helps. My system you can see above, but KODI is now updatet to 19.1
Thanks in advance.

**Comment by basilgello** [comment]
@AldaTheFirst It is unrelated to the initial bug. I assume you are talking about the `HH:MM` indicator in player window? If yes, `00:04 / 00:03` is temporary due to buffering. Do I understand you correctly?

If no, please post screenshot.

**Comment by AldaTheFirst** [comment]
reading your post again and reading something about "due of buffering" I noticed, that I swapped seconds and minutes in my explanation. Here is an correction:

For example:
The recording started at 2:00:00 p.m. and remains active all the time
Playback starts at 2:03:00 p.m.
The display shows 00:00 / 03:00    (I assume that 03 are the minutes and 00 are the seconds, that means, recording is 3 minutes)
After 2 minutes the display shows 02:00 / 03:00, but should show 02:00 / 05:00
After another 2 minutes the display shows 04:00 / 03:00, means I saw more than the duration of the recording ? Could not be. Should show 04:00 / 07:00
Pressing Jump Forward (ff) will reset the playback to 03:00 and start playing from that point.
If you stop and restart playback, recording will start from the beginning of 00:00

**Comment by basilgello** [comment]
@AldaTheFirst First @phunkyfish and I thought you are running vanilla Debian bullseye not OSMC (which is a derivative from buster!) and there is something to do with my patch for the Debian packaging. Now I see it is not the case.

What probably IS the cause is "GetStreamTimes: VNSI… returned:server error". AFAIK GetStreamTimes especially returns the current playback.
Since @FernetMenta is no more maintaining VNSI plugin, I think @ksooo might have an idea what's happening on VDR-VNSI side.

**Comment by ksooo** [comment]
> I think @ksooo might have an idea what's happening on VDR-VNSI side.

 soory, no. i don't use or maintain vdr vnsi

**Comment by AldaTheFirst** [comment]
Any news here? I can live with this error, but it is still quite unpleasant. If you accidentally fast forward too far, playback stops and when you start again you have to search for the point where you stopped watching. There is no way to go back to the last point you watched except to search manually. Would be nice to have a solution.

**Comment by ksooo** [comment]
Honestly, as nobody is maintaining the vdr.vnsi addon and vnsi backend for years now, I think it's time to think about switching to another PVR backend (or frotend, if you want to stick to vdr).

**Comment by AldaTheFirst** [comment]
Thanks for your answer. I am afraid I have to switch. The Installation ist only about 20 years old. What a pity... :-)

Am 10. August 2024 10:38:17 MESZ schrieb Kai Sommerfeld ***@***.***>:
